src/prompt_file.py

```diff
@@ -1,262 +1,164 @@
-            with Path(save_path).open('a') as f:
-                f.writelines(lines)
-        
-        print('\n#   Context saved to {}'.format(save_name))
+        self.prompt_engine.remove_last_interaction()
     
     def start_multi_turn(self):
         """
         Turn on context mode
         """
-        self.config['multi_turn'] = 'on'
-        self.set_config(self.config)
-        print("\n#   Multi turn mode is on")
+        self.prompt_engine.config.model_config.multi_turn = "on"
 
     
     def stop_multi_turn(self):
         """
         Turn off context mode
         """
-        self.config['multi_turn'] = 'off'
-        self.set_config(self.config)
-        print("\n#   Multi turn mode is off")
+        self.prompt_engine.config.model_config.multi_turn = "off"
     
     def default_context(self):
         """
         Go to default context
         """
-        self.load_context(self.context_source_filename)
+        shell_context_path = Path(os.path.join(os.path.dirname(__file__), "..", "contexts", f"{self.prompt_engine.config.model_config.shell}-context.yaml"))
+        self.prompt_engine = self.create_prompt_engine_from_yaml(shell_context_path)
+        self.save_prompt_engine(self.prompt_engine, self.default_config_path)
+        print (f'\n#   Switched to f"{self.prompt_engine.config.model_config.shell}-context.yaml')
+
+    def clear_context(self):
+        shell_context_path = Path(os.path.join(os.path.dirname(__file__), "..", "contexts", f"{self.prompt_engine.config.model_config.shell}-context.yaml"))
+        self.prompt_engine = self.create_prompt_engine_from_yaml(shell_context_path)
+        self.save_prompt_engine(self.prompt_engine, self.default_config_path)
+        print (f'\n#   Cleared context')
+
+    def save_to(self, filename):
+        if not filename.endswith('.yaml'):
+            filename = filename + '.yaml'
+        filepath = Path(os.path.join(os.path.dirname(__file__), "..", "contexts", filename))
+        with open(filepath, 'w') as f:
+            f.write(self.prompt_engine.save_yaml())
     
-    def load_context(self, filename, initialize=False):
+    def load_context(self, filename):
         """
         Loads a context file into current_context
         """
-        if not filename.endswith('.txt'):
-            filename = filename + '.txt'
+        if not filename.endswith('.yaml'):
+            filename = filename + '.yaml'
         filepath = Path(os.path.join(os.path.dirname(__file__), "..", "contexts", filename))
 
         # check if the file exists
         if filepath.exists():
-            with filepath.open('r') as f:
-                lines = f.readlines()
             
             # read in the engine name from openaiapirc
             config = configparser.ConfigParser()
             config.read(API_KEYS_LOCATION)
-            ENGINE = config['openai']['engine'].strip('"').strip("'")
+            engine = config['openai']['engine'].strip('"').strip("'")
```

Comment:
Although the engine is present in the yaml file, this is the local openairpc file that gets created. This can override the yaml file, and for that strip is needed